File: ArkhamAsylum.Lib/Models/Bed.cs
using Newtonsoft.Json;
using System;

namespace ArkhamAsylum.Lib.Models
{
    public class Bed : Entity
    {
        public string Code { get; set; }

        public Guid RoomId { get; set; }

        [JsonIgnore]
        public Room Room { get; set; }

        public int RoomNumber
        {
            get
            {
                return Room.Number;
            }
        }
    }
}

File: ArkhamAsylum.Web/wwwroot/Scripts/Models/Bed.js
class Bed extends Entity
{
    constructor(json)
    {
        super(json)
        if (json)
        {
            this.Code = json.code;
            this.RoomId = json.roomId;
        }
        else
        {
            this.Code = "";
            this.RoomId = "00000000-0000-0000-0000-000000000000"
        }
    }
}